### DISCUSSION TITLE
Only update package 'X' if there is also an update for package 'Y'

### DISCUSSION THREAD

**Discussion by JoostvdB94** [renovatebot/renovate#24826] [Suggest an Idea]
### What would you like Renovate to be able to do?

I would like to specify rules for a group of dependencies in a way that an update would only be suggested if another package is also updated.

In the current situation it might be possible that the following scenario occurs:
- OpenSearch is updated (container / docker)
- The Helm template for the new OpenSearch is not yet available (helm)

When running Renovate, we I would like to be notified of a new OpenSearch update only when a new version of Helm (targetting that updated image) is also available. 

The current grouping functionality makes a PR when at least 1 of the artifacts matching the criteria is updated. Could there also be a way to specify that there should only be a single PR when both OpenSearch and its Helm template are updated?

### If you have any ideas on how this should be implemented, please tell us here.

Add a packageRule where a condition can be specified that checks other package upgrades (within the same group)

I have found something that might already be implemented, but the documentation about this is quite unclear:
- `matchDepName`
- `matchDepPatterns`

Which, by the name, might be used to specify dependencies for the group? Or am I mistaken?

### Is this a feature you are interested in implementing yourself?

No

**Comment by viceice** [comment]
This isn't possible yet and the mentioned fields do work like the `packageName` fields.

**Comment by viceice** [comment]
I like that idea in general, but this seems to be very complicated to implement

**Comment by rarkins** [comment]
A way of solving this would be something like a generic "minimumUpdates" field for groups where the branch isn't created unless at least that number is met. Eg 2

**Comment by JoostvdB94** [comment]
@rarkins Thanks for your suggestion! That would fix 90% of the problems and indeed sounds much simpler.

However, it does not work so well for groups that are made based on patterns. In that case you cannot know the number of other updates. 

Let's imagine the following scenario
We have applications **A**, **B**, **C** and **D**

Applications **B**, **C** and **D** all use resources exposed by application **A**. 

You could specify the minimum number of updates to 2 and hope that when **B** changes, **A** changes too. 
But when B and C update but A is unchanged, you would still get a PR suggesting an upgrade.

A more deterministic approach would be to specify package dependencies. 
**B** depends on **A**
**C** depends on **A**
**D** depends on **A**

But if we can count the number of package updates in a group, we should also be able to see and check what packages are in that group, right?

**Comment by rarkins** [comment]
What percentage would you reach if you also combine with minimumReleaseAge setting? Eg wait a day or more before raising the PR. How often would A trail B,C,D?

**Comment by StefanLobbenmeierObjego** [comment]
For my use case, I would prefer a "compatability" settings to support this. This should be an enum of "MAJOR", "MINOR", "PATCH" and will indicate what version compatability is required across grouped packages.

Something like this:
- ✅ PR is created
- 🔴 No PR is created, wait for compatible versions


<table>
<th>
 <td>Version A
 <td>Version B
 <td>MAJOR
 <td>MINOR
 <td>PATCH
 <td>default
<tr>
 <td>Major version is different
 <td>2.5
 <td>3.4
 <td>🔴 
 <td>🔴 
 <td>🔴 
 <td>✅
<tr>
 <td>Minor version is different
 <td>2.5
 <td>2.7
 <td>✅
 <td>🔴 
 <td>🔴 
 <td>✅
<tr>
 <td>Patch version is different
 <td>2.5.3
 <td>2.5.4
 <td>✅
 <td>✅ 
 <td>🔴 
 <td>✅
<tr>
 <td>Versions are exactly the same
 <td>2.5.3
 <td>2.5.3
 <td>✅
 <td>✅ 
 <td>✅ 
 <td>✅
</table>


As a first attempt, we could just look at the latest version, but ideally we check all versions and try to find the latest compatible versions.

**Reply by StefanLobbenmeierObjego** [reply]
my usecase in more detail:
ironpdf java client library udpates once a month, requires the latest version of the docker container at the time of release, usually minor version is the same

ironpdf docker container updates 20 times a month, I am not interested in most of those updates since they are not compatible with the java client library

**Comment by felipecrs** [comment]
I believe this can be done by grouping both dependencies and setting [`minimumGroupSize: 2`](https://docs.renovatebot.com/configuration-options/#minimumgroupsize).

**Reply by StefanLobbenmeierObjego** [reply]
True, looks like the original idea by @rarkins here https://github.com/renovatebot/renovate/discussions/24826#discussioncomment-7155575 for was implemented and released last August: https://github.com/renovatebot/renovate/issues/37218#issuecomment-3184521873
